qutebrowser/misc/sql.py

```diff
@@ -475,3 +476,87 @@ def select(self, sort_by, sort_order, limit=-1):
                           sort_order=sort_order))
         q.run(limit=limit)
         return q
+
+
+class Database:
+
+    _database: Optional[QSqlDatabase] = None
+    _user_version: Optional[UserVersion] = None
```

Comment:
By setting values here, you're making them class variables rather than instance variables - and also they need to be `Optional` that way despite being set to something in `__init__`.

You could set the type here (without the `= None`), but I'd suggest to just remove those two lines, and add type annotations in `__init__` if mypy requires it.

Reply:
Oh, sure, as long as mypy is smart enough to know that `self._database` exists if I only set it in `__init__`.

Reply:
It should be! It's the usual way to create [instance variables (rather than class variables)](https://www.digitalocean.com/community/tutorials/understanding-class-and-instance-variables-in-python-3) in Python.